msprime/simulations.py

```diff
@@ -175,6 +175,7 @@ def simulator_factory(
         length=None,
         recombination_rate=None,
         recombination_map=None,
+        finite_recombination_sites=False,
```

Comment:
Let's remove this parameter for now and have a think about what we want the API to be.